src/meshlink.c

```diff
@@ -1790,6 +1787,53 @@ meshlink_node_t **meshlink_get_all_nodes_by_dev_class(meshlink_handle_t *mesh, d
 	return result;
 }
 
+static bool search_node_by_dev_class(node_t *node, void *condition) {
+	dev_class_t *devclass = (dev_class_t *)condition;
+
+	if(*devclass == node->devclass) {
+		return true;
+	}
+
+	return false;
+}
+
+static bool search_node_by_submesh(node_t *node, void *condition) {
+	if(condition == node->submesh) {
+		return true;
+	}
+
+	return false;
+}
+
+meshlink_node_t **meshlink_get_all_nodes_by_dev_class(meshlink_handle_t *mesh, dev_class_t devclass, meshlink_node_t **nodes, size_t *nmemb) {
+	if(!mesh || ((int)devclass < 0) || (devclass > _DEV_CLASS_MAX) || !nmemb) {
+		meshlink_errno = MESHLINK_EINVAL;
+		return NULL;
+	}
+
+	return meshlink_get_all_nodes_by_condition(mesh, &devclass, nodes, nmemb, search_node_by_dev_class);
+}
+
+meshlink_node_t **meshlink_get_all_nodes_by_submesh(meshlink_handle_t *mesh, meshlink_submesh_t *submesh, meshlink_node_t **nodes, size_t *nmemb) {
+	meshlink_submesh_t *s = NULL;
+
+	if(!mesh || !submesh || !submesh->name || !nmemb) {
```

Comment:
In fact, since we don't care about `submesh->name`, you can drop that check from this if-statement.

Reply:
I removed submesh->name parameter from the condition check